docs/fundamentals/networking/http/httpclient-migrate-from-httpwebrequest.md

```diff
@@ -0,0 +1,468 @@
+
+**Old Code Using `HttpWebRequest`**:
+
+```csharp
+ServicePointManager.DnsRefreshTimeout = 60000;
+ServicePointManager.EnableDnsRoundRobin = true;
+HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
+HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+```
+
+**New Code Using `HttpClient`**:
+
+```csharp
+ConcurrentDictionary<string, DnsCache> dnsCache = new(StringComparer.OrdinalIgnoreCase);
+
+TimeSpan dnsRefreshTimeout = TimeSpan.FromMinutes(1);
+var handler = new SocketsHttpHandler
+{
+    ConnectCallback = async (context, cancellationToken) =>
+    {
+        string host = context.DnsEndPoint.Host;
+        
+        DnsCache? cacheItem;
+        if (!dnsCache.TryGetValue(host, out cacheItem) || DateTime.Now > cacheItem.CacheExpireTime)
+        {
+            dnsCache.TryRemove(host, out _);
+            cacheItem = new DnsCache(await Dns.GetHostAddressesAsync(host, cancellationToken), DateTime.Now.Add(dnsRefreshTimeout), 0);
+            dnsCache.TryAdd(host, cacheItem);
+        }
+
+        IPAddress connectAddress;
+        // If there is only one address, we don't need to do anything special
+        if (cacheItem.Addresses.Length == 1)
+        {
+            connectAddress = cacheItem.Addresses[0];
+        }
+        else
+        {
+            int index = cacheItem.Index;
+            connectAddress = cacheItem.Addresses[index];
+            cacheItem.IncreaseIndex();
+        }
+
+        var socket = new Socket(SocketType.Stream, ProtocolType.Tcp)
+        {
+            NoDelay = true
+        };
+        try
+        {
+            await socket.ConnectAsync(connectAddress, context.DnsEndPoint.Port, cancellationToken);
+            return new NetworkStream(socket, ownsSocket: true);
+        }
+        catch
+        {
+            socket.Dispose();
+            throw;
+        }
+    }
+};
+var client = new HttpClient(handler);
+var response = await client.GetAsync(uri);
+
+class DnsCache(IPAddress[] addresses, DateTime cacheExpireTime, int index)
+{
+    public IPAddress[] Addresses { get; } = addresses;
+    public DateTime CacheExpireTime { get; } = cacheExpireTime;
+    public int Index { get; private set; } = index;
+
+    public void IncreaseIndex()
+    {
+        Index = (Index + 1) % Addresses.Length;
+    }
+}
+```
+
+### Example: Setting SocketsHttpHandler Properties
+
+TODO:
+
+### Example: Change ImpersonationLevel
+
+TODO:
+
+## Usage of Certificate and TLS Related Properties in HttpClient
+
+When working with `HttpClient`, you may need to handle client certificates for various purposes, such as custom validation of server certificates or fetching the server certificate. `HttpClient` provides several properties and options to manage certificates effectively.
+
+### Example: Enabling CRL Check with SocketsHttpHandler
```

Comment:
NIT: I don't like using abbreviations before they are introduced, so I'd opt to something like:

Check certificate revocation list with `SocketsHttpHandler`